### ISSUE TITLE
[Question] How to select materialUI autocomplete list item?

### ISSUE THREAD

**Issue by dadamsGS** [microsoft/playwright#13451]
This is a follow up to: https://github.com/microsoft/playwright/issues/8397

I am trying to automate a form that has a [MaterialUI Autocomplete](https://mui.com/material-ui/react-autocomplete/) element. The list of items that are supposed to displayed take a couple seconds to load and populate the list items from which I want to select by value.

My issue is that the list of items starts to open and then immediately closes which does not build the list of select options and times out my test. When I debug and step through the test, the list opens and selects as intended.

I attached a screen capture:
https://user-images.githubusercontent.com/93952012/162583054-7113e3a0-7671-49da-8f7b-b11d5bb7543d.mov

I am also attaching the [trace.zip](https://github.com/microsoft/playwright/files/8457366/trace.zip). From the trace I can see that the request for data to populate the list is returning the correct data.

My test is really simple right now (I find that React selects are one of the more challenging aspects for automation):
```
test.describe('IBT demo', () => {
  test('verify search functionality', async ({ page }) => {
    await page.goto('/app/purchases/ticket-library/')
    await page.click('#create-inbound-ticket-button')

    let idValue = await page.locator(`div[name=customer] label`).getAttribute('id')
    idValue = idValue.match(/(^mui-[0-9]*)/)[1]
    
    await page.click('div[name=customer]', { timeout: 6000 })
    await page.click(`#${idValue}-option-8`, { timeout: 6000 })

    await page.waitForSelector(`//ul[@id='${idValue}-listbox']/li`)
  });
});
```

Any guidance on this would be helpful. I would like to transition from Cypress.

**Comment by dadamsGS** [comment]
As a quick follow up, I have tried mouse interactions (up/down, click); does not open any drop-down

**Comment by mxschmitt** [comment]
I tried [codegen](https://playwright.dev/docs/next/cli#generate-code) which was generating the following for me:

```js
  // Click text=MovieMovie >> div >> nth=0
  await page.locator('text=MovieMovie >> div').first().click();

  // Click text=12 Angry Men
  await page.locator('text=12 Angry Men').click();
```

it was working fine!

I'd strongly recommend to use Locators as well instead of page.* actions see [here](https://playwright.dev/docs/next/locators).

**Comment by dadamsGS** [comment]
Thanks for the response and the example.

I did try `codegen` as well to get the selectors. That was another attempt I didn't mention in my posting.

Looking at your example (https://mui.com/material-ui/react-autocomplete/), those autocomplete dropdowns have the data already on the page and build the dropdown immediately. 

The implementation I am testing seems to not build the autocomplete immediately. One first page load, when opening the dropdown I see "No options", then "Loading...", then the options. 
When I refresh the page, when loading the dropdown I see "No options", then the options populate. I wouldn't say the options list is massive (~100 options). I'll check with the devs to see if there is a way to load the dropdown without the additional steps.


UPDATE: I tried to open the autocomplete twice in a row hoping for the option data to cache somewhere and it looks like it's working.
Something like:
```
test.describe('IBT demo', () => {
  test('verify search functionality', async ({ page }) => {
    await page.goto('/app/purchases/ticket-library/')
    await page.click('#create-inbound-ticket-button')
    
    await page.click('div[name=customer]')
    await page.click('div[name=customer]')

    let idValue = await page.locator(`div[name=customer] label`).getAttribute('id')
    idValue = idValue.match(/(^mui-[0-9]*)/)[1]

    await page.click(`#${idValue}-option-8`, { timeout: 6000 })

    await page.waitForSelector(`//ul[@id='${idValue}-listbox']/li`)
  });
});
```
Not idea but it's working. I can work with the devs on better timing for those options.
